bin/cylc-ref-graph

```diff
@@ -0,0 +1,240 @@
+    nodes = (
+        node
+        for left, right, _, suicide, _ in graph
+        for node in (left, right)
+        if node
+        if show_suicide or not suicide
+    )
+    for node in sorted(set(nodes), key=node_sort):
+        print('node "%s" "%s"' % (node, node.replace('.', r'\n')))
+
+    print('stop')
+
+
+def graph_inheritance(config):
+    """Implement ``cylc-graph --reference --namespaces``."""
+    edges = set()
+    nodes = set()
+    for namespace, tasks in config.get_parent_lists().items():
+        for task in tasks:
+            edges.add((task, namespace))
+            nodes.add(task)
+
+    for namespace in config.get_parent_lists():
+        nodes.add(namespace)
+
+    for edge in sorted(edges):
+        print('edge %s %s' % edge)
+
+    print('graph')
+
+    for node in sorted(nodes):
+        print('node %s %s' % (node, node))
+
+    print('stop')
+
+
+def get_config(suite, opts, template_vars=None):
+    """Return a SuiteConfig object for the provided reg / path."""
+    try:
+        suiterc = SuiteSrvFilesManager().get_suite_rc(suite)
+    except SuiteServiceFileError:
+        # could not find suite, assume we have been given a path instead
+        suiterc = suite
+        suite = 'test'
+    return SuiteConfig(suite, suiterc, opts, template_vars=template_vars)
+
+
+def get_option_parser():
+    """CLI."""
+    parser = COP(
+        __doc__, jset=True, prep=True,
+        argdoc=[
+            ('[SUITE]', 'Suite name or path'),
+            ('[START]', 'Initial cycle point '
+             '(default: suite initial point)'),
+            ('[STOP]', 'Final cycle point '
+             '(default: initial + 3 points)')])
+
+    parser.add_option(
+        '-u', '--ungrouped',
+        help='Start with task families ungrouped (the default is grouped).',
+        action='store_true', default=False, dest='ungrouped')
+
+    parser.add_option(
+        '-n', '--namespaces',
+        help='Plot the suite namespace inheritance hierarchy '
+             '(task run time properties).',
+        action='store_true', default=False, dest='namespaces')
+
+    parser.add_option(
+        '-r', '--reference',
+        help='Output in a sorted plain text format for comparison purposes.',
+        action='store_true', default=True, dest='reference')
+
+    parser.add_option(
+        '--show-suicide',
+        help='Show suicide triggers.  They are not shown by default, unless '
+             'toggled on with the tool bar button.',
+        action='store_true', default=False, dest='show_suicide')
+
+    parser.add_option(
+        '--icp', action='store', default=None, metavar='CYCLE_POINT', help=(
+            'Set initial cycle point. Required if not defined in suite.rc.'))
+
+    return parser
+
+
+def main(opts, suite=None, start=None, stop=None):
+    """Implement ``cylc graph``."""
+    if opts.ungrouped and opts.namespaces:
+        raise Exception('Cannot combine --ungrouped and --namespaces.')
+    if not opts.reference:
```

Comment:
Currently this means only ``cylc ref-graph --reference``  works, but I think we want instead a plain ``cylc ref-graph``, which assumes the reference behaviour anyway. I think we should (& probably intended to) drop the ``--reference`` option?

Reply:
That was deliberate in case users tried to replicate the command line of `cylc graph --reference` exactly - but I did change the flag default to `True` which makes its use optional. I have now added a comment to the command help to say that it is optional.  Is that OK?

Reply:
Ah, I see, in that case that is completely fine (even without the note, but it is helpful so it's good that it has been added). That motivation did not occur to me, but it is a wise idea.